src/main/java/org/alfresco/heartbeat/HeartBeatJob.java

```diff
@@ -0,0 +1,209 @@
+/*
+ * #%L
+ * Alfresco Repository
+ * %%
+ * Copyright (C) 2005 - 2017 Alfresco Software Limited
+ * %%
+ * This file is part of the Alfresco software.
+ * If the software was purchased under a paid Alfresco license, the terms of
+ * the paid license agreement will prevail.  Otherwise, the software is
+ * provided under the following open source license terms:
+ *
+ * Alfresco is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Alfresco is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Alfresco. If not, see <http://www.gnu.org/licenses/>.
+ * #L%
+ */
+package org.alfresco.heartbeat;
+
+import org.alfresco.heartbeat.datasender.HBData;
+import org.alfresco.heartbeat.datasender.HBDataSenderService;
+import org.alfresco.repo.lock.JobLockService;
+import org.alfresco.repo.lock.LockAcquisitionException;
+import org.alfresco.service.namespace.NamespaceService;
+import org.alfresco.service.namespace.QName;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.quartz.Job;
+import org.quartz.JobDataMap;
+import org.quartz.JobExecutionContext;
+import org.quartz.JobExecutionException;
+
+import java.util.List;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+/**
+ * The scheduler job responsible for triggering a heartbeat on a regular basis.
+ */
+
+public class HeartBeatJob implements Job
+{
+    /** The logger. */
+    private static final Log logger = LogFactory.getLog(HeartBeatJob.class);
+
+    /** Time to live 5 seconds */
+    private static final long LOCK_TTL = 5000L;
+    /** Additional 5 seconds for how much longer the log will be kept */
+    private static final long LOCK_TTL_OFFSET = 5000L;
+
+    public void execute(final JobExecutionContext jobexecutioncontext) throws JobExecutionException
+    {
+        final JobDataMap dataMap = jobexecutioncontext.getJobDetail().getJobDataMap();
+        final HBBaseDataCollector collector = (HBBaseDataCollector) dataMap.get("collector");
```

Comment:
The keys in the Map ("collector", "hbDataSenderService" and "jobLockService") should be defined as ```public static final String``` fields on this class, and then referenced from the HBDataCollectorServiceImpl.java. Otherwise, it is easy to forget one or introduce a typo when changing names.